src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardSamlBundle/Security/Authentication/Provider/SamlProvider.php

```diff
@@ -87,22 +87,24 @@ public function authenticate(TokenInterface $token)
     {
         $translatedAssertion = $this->attributeDictionary->translate($token->assertion);
 
-        $nameId         = $translatedAssertion->getNameID();
-        $email          = $this->getSingleStringValue('mail', $translatedAssertion);
-        $commonName     = $this->getSingleStringValue('commonName', $translatedAssertion);
+        $nameId = $translatedAssertion->getNameID();
+        $email = $this->getSingleStringValue('mail', $translatedAssertion);
+        $commonName = $this->getSingleStringValue('commonName', $translatedAssertion);
 
         // Default to the ROLE_USER role for services.
         $role = 'ROLE_USER';
 
         try {
             // An exception is thrown when isMemberOf is empty.
-            $teamNames = (array) $translatedAssertion->getAttributeValue('isMemberOf');
+            $teamNames = (array)$translatedAssertion->getAttributeValue('isMemberOf');
         } catch (RuntimeException $e) {
             $teamNames = [];
         }
 
-        if (in_array($this->administratorTeam, $teamNames)) {
-            $role = 'ROLE_ADMINISTRATOR';
+        foreach ($this->administratorTeams as $teamName) {
+            if (in_array($teamName, $teamNames)) {
+                $role = 'ROLE_ADMINISTRATOR';
```

Comment:
Maybe break the loop here? Or use some fancy-the-pancy array function, could `array_intersect` help?